tools/pipelines/templates/build-npm-package.yml

```
@@ -620,3 +628,38 @@ stages:
         namespace: ${{ parameters.namespace }}
         tagName: ${{ parameters.tagName }}
         publishNonScopedPackages: ${{ variables.publishNonScopedPackages }}
+
+  # Capture per-pipeline stage results
+  - stage: runAfterAll
+    displayName: Retrieve and upload client stage stats to Kusto
+    condition: and(succeededOrFailed(), ne(variables['Build.Reason'], 'PullRequest'), eq(${{parameters.telemetry}}, true))
+    dependsOn: 
+      - build
+
+    jobs:
+      - template: include-test-perf-benchmarks.yml
+        parameters:
+            testWorkspace: ${{ variables.testWorkspace }}
+            loggerPackage: ${{ variables.loggerPackage }}
+            poolBuild: Small
+            customSteps:
+
+            - task: Bash@3
+              displayName: Retrieve buildId results
+              inputs:
+                targetType: 'inline'
+                workingDirectory: $(toolAbsolutePath)
+                script: |
+                  echo "creating output folder"
+                  mkdir -p ${{ variables.testWorkspace }}/timingOutput
+                  echo "Executing curl command ..."
+                  echo  'curl -u ":<REDACTED>" "https://dev.azure.com/fluidframework/internal/_apis/build/builds/$(Build.BuildId)/timeline"'
+                  curl -u ":$(System.AccessToken)" "https://dev.azure.com/fluidframework/internal/_apis/build/builds/$(Build.BuildId)/timeline\?api-version\=6.0-preview.1" > ${{ variables.testWorkspace }}/timingOutput/output.json
```

The place where you copied this from should have done it this way (it actually doesn't even use the `ADO_API_TOKEN` variable it defines). Will make it easier to extract this bash script to a file at some point. I think this will work as is, but let's double check if I got the syntax right.
```suggestion
                  echo  'curl -u ":<REDACTED>" "https://dev.azure.com/fluidframework/internal/_apis/build/builds/$BUILD_ID/timeline"'
                  curl -u ":$ADO_API_TOKEN" "https://dev.azure.com/fluidframework/internal/_apis/build/builds/$BUILD_ID/timeline\?api-version\=6.0-preview.1" > ${{ variables.testWorkspace }}/timingOutput/output.json
```